bundles/org.eclipse.core.runtime/src/org/eclipse/core/runtime/Platform.java

```diff
@@ -1462,4 +1464,47 @@ public static boolean inDebugMode() {
 	public static boolean inDevelopmentMode() {
 		return PlatformActivator.getContext().getProperty("osgi.dev") != null; //$NON-NLS-1$
 	}
+
+	/**
+	 * Retrieves the <b>native</b> system encoding ({@link Charset}) based on the
+	 * system locale.
+	 * <p>
+	 * In case the detection of native encoding fails, returns
+	 * {@link Charset#defaultCharset()}, which is always {@code UTF-8} on Java 18
+	 * and later.
+	 *
+	 * @return the {@link Charset}, never null
+	 * @see <a href="https://openjdk.java.net/jeps/400">JEP 400</a>
+	 * @since 3.26
+	 */
+	public static Charset getNativeEncoding() {
+		Charset result = defaultCharset;
```

Comment:
I know for certain users that use `-Dfile.encoding=UTF-8` in their eclipse.ini to e.g always enforce UTF-8 even on windows, but don't know if we can have values from eclipse ini or if it matters breaking those... maybe we can first check a value of `eclipse.encoding` just in case?

Reply:
"file.encoding" is not a solution but more the problem in the usecase of https://github.com/eclipse-jdt/eclipse.jdt.debug/pull/47